cypress/integration/apprenticeFilterSpec.js

```diff
@@ -0,0 +1,75 @@
+/// <reference types="cypress" />
+
+describe('Should filter Apprenticeships correctly', () => {
+  beforeEach(() => {
+    cy.visit('http://localhost:3000/')
+  })
+
+  it('filters all', function() {
+    cy.get(`#filters-all`).click();  
+    cy.get('.list__item__content-category').contains('software')
+  });
+
+  const filters = ['development', 'design', 'software', 'mobile']
+  filters.map(filter => {
+    return it(`filters ${filter}`, function() {
+      cy.get(`#filters-${filter}`).click()
```

Comment:
Cool idea to build the tests based on the `filters` array! A few suggestions:

- Since we don't need a return value from the `map`, we can switch it to a `forEach` and remove the `return`
- Testing that the list retains the correct elements is a good start, but I think we need to test that filtered items are not visible. One helpful way to "test your tests" is to comment out an action that you'd expect to cause a failure and see how the test responds. In this case, if we comment out the filter `click`, the tests will still pass because nothing is filtered from the list. We could add something like this, that checks that the other elements in the array are not visible:

```
      // assert that only the selected filter displays
      filters.filter(f => f !== filter).forEach((filter) => {
        cy.get('.list__item__content-category:visible').should('not.contain', filter)
      })
```
☝️ I had to add `:visible` to the selector there to get it to filter the hidden items, not sure offhand but there may be a better way to do that

Reply:
Good thinking! I looked at a few different approaches to filter hidden items as well like checking the DOM element's `offsetParent` or getting the style attribute but the `:visible` selector looks like one of the better options to go with.